pkgs/stdenv/generic/setup.sh

```diff
@@ -1021,6 +1021,21 @@ configurePhase() {
             echo "fixing libtool script $i"
             fixLibtool "$i"
         done
+
+        # replace `/usr/bin/file` with `file` in any `configure`
+        # scripts with vendored libtool code.  Preserve mtimes to
+        # prevent some packages (e.g. libidn2) from spontaneously
+        # autoreconf'ing themselves
```

Comment:
The latest version of libtool already has this fixed by our very own @aakropotkin, and I would say that when a package has pre-generated build scripts using ancient version of `libtool`, it likely uses other old macros as well so this might not be the only issue. Did you consider just forcing building for all packages, e.g. by running `make distclean`?

Though, I recognize that this might be hard due to bootstrapping, removing files that we are not able to generate from the tarball, different projects having [different `autogen.sh` styles](https://www.sourceware.org/autobook/autobook/autobook_43.html), not supporting autoreconf, so we might still want to go with this for the time being.

Reply:
Actually, just remembered Debian has that as policy so we could look there: https://wiki.debian.org/Autoreconf

Reply:
> The latest version of libtool already has this fixed by our very own @aakropotkin

Yes I know, I wrote [the PR](https://github.com/NixOS/nixpkgs/pull/167071) that bumped the version in nixpkgs!

It is wonderful that libtool 2.4.7 no longer forces `/usr/bin/file` to be hardcoded.  Unfortunately it does not fix the entire problem.  Here are the two remaining failure modes:

1. If upstream authors set `FILECMD` when they run the autotools, whatever they set it to will get hardcoded into the vendored source tarball.
2. If the source tarball was produced with libtool<=2.4.6.  This is 100% of all upstream software released before 2022-Mar-17.

> ancient version of `libtool`

I guess if "ancient" means "more than [two months old](https://savannah.gnu.org/forum/forum.php?forum_id=10139)"...

> Did you consider just forcing building for all packages, e.g. by running make distclean?

If you mean adding `autoreconfHook` then yes, that was one of the first things I tried.  Unfortunately if you blindly `autoreconfHook` every package you get tons of failures because they only work when `autoreconf`ed with exactly the version that the original author used.  Here is [a great example of the breakage](https://github.com/NixOS/nixpkgs/pull/163401#issuecomment-1072040524).